tests/test_numpy_wrapper.py

```diff
@@ -350,6 +350,58 @@ def test_binary_operator_nontrainable(self):
         assert isinstance(res, np.tensor)
         assert not res.requires_grad
 
+    def test_binary_operator_mixed_trainable_left(self):
+        """Test that binary operators on one trainable and one non-trainable
+        arrays result in trainable output."""
+        x = np.array([[1, 2], [3, 4]], requires_grad=True)
+        y = np.array([[5, 6], [7, 8]], requires_grad=False)

```

Comment:
So the elements inside a trainable array will become trainable when indexed like `x[0]`, right? Is this why we only need to check that the overall output array is trainable?

I'm thinking of an edge case `x =  np.array([np.array([1, 2], requires_grad=False), [3, 4]], requires_grad=True)` here

Reply:
Should do!

![image](https://user-images.githubusercontent.com/2959003/98097235-b17cbf80-1ec7-11eb-91ad-8d8b743b1699.png)